sql/src/test/java/io/crate/integrationtests/PostgresITest.java

```diff
@@ -507,19 +507,11 @@ public void testExecuteBatchWithOneFailingAndNothingExecuted() throws Exception
             preparedStatement.setString(1, Integer.toString(3));
             preparedStatement.addBatch();
 
-            try {
-                preparedStatement.executeBatch();
-                fail("executeBatch must throw an exception");
-            } catch (BatchUpdateException e) {
-                assertThat(e.getUpdateCounts(), is(new int[]{-3, -3, -3}));
-                SQLException nextException = e.getNextException();
-                assertThat(nextException.getMessage(), Matchers.containsString("Cannot cast 'foobar' to type integer"));
-            }
+            assertThat(preparedStatement.executeBatch(), is(new int[]{-3, -3, -3}));
```

Comment:
This is a behavior change. Shouldn't we still be throwing an exception?

Reply:
Previously, this exception was thrown due an exception in insert from values analyzer and then caught in `PostgresWireProtocol` so it was not added to the job logs due the way we handle the analyzer exceptions in `execBulk` or not handle. After the move to the unbound analyzer the error handling has changed a bit, now such statement won't fail at analysis, but at execution. So the job log entry will be added here https://github.com/crate/crate/pull/9465/files#diff-8df7e31e8934a6bebea6f7bf4532b50cR510 and then if we throw an exception as it was before, this job log won't be updated as ended or smth. Therefore, i've implemented handling exception in such case like this using futures https://github.com/crate/crate/pull/9465/files#diff-a74268cd5e9c526d14e66b3dc5daa81aR366-R370

Yeah, we do not throw the exception now, but the result is still the same, nothing is inserted and the statement still fails on handler.

Reply:
It's a subtle change in how users have to handle errors in their applications. I think we should at least have an entry in the release notes for this.

Reply:
I'll add an entry then.